### DISCUSSION TITLE
Is this library able to collect SMTP traffic infos?

### DISCUSSION THREAD

**Discussion by Dennis-Zhang-SH** [wneessen/go-mail#134] [Q&A] [answered]
I had a app sending emails based on python, I want to switch to golang but a lot of libraries lack of transport infos.

**Comment by wneessen** [comment]
Hi @Dennis-Zhang-SH,

glad you're interested in using go-mail. The package does indeed allow for transport logging. Since v0.3.9, we've added a debug logging option which allows you to see the communication between the server and the client. Some details and an example screenshot can be found in the PR if that helps: https://github.com/wneessen/go-mail/pull/102

**Comment by Dennis-Zhang-SH** [comment]
> Hi @Dennis-Zhang-SH,
> 
> glad you're interested in using go-mail. The package does indeed allow for transport logging. Since v0.3.9, we've added a debug logging option which allows you to see the communication between the server and the client. Some details and an example screenshot can be found in the PR if that helps: #102

Can this log info been collected as string? I may need to save it in the database.

**Reply by wneessen** [reply, marked as answer]
The debug logging feature implements a simple logging interface, so you can provide a custom logger instead of the provided simple logger, we're shipping with the package. As long as your custom logger implements that interface, you should be ready to go. Just build a logging function that takes the provided string and stores it in a database.

**Comment by Dennis-Zhang-SH** [comment]
> The debug logging feature implements a simple logging interface, so you can provide a custom logger instead of the provided simple logger, we're shipping with the package. As long as your custom logger implements that interface, you should be ready to go. Just build a logging function that takes the provided string and stores it in a database.

Oh okay just found it, that would be great.
